### DISCUSSION TITLE
SSH config file parameters

### DISCUSSION THREAD

**Discussion by tony1661** [grafviktor/goto#73] [Q&A] [answered]
Hi,
I tried out your app and I love it so far. 

I have a question for you. Is it possible to have ssh options passed into the hosts? For example, I have some old legacy gear that needs `HostkeyAlgorithms +ssh-dss,ssh-rsa` to be set. I know that I can set this in my config file for SSH but it'd be nice to have it all in one place.

**Comment by grafviktor** [comment]
Hi @tony1661 . Thank you very much for your interest! 

I think, it should work as GoTo is just a wrapper. Please try to define a host with the same alias in GoTo as you have it in your ssh config:

<img width="260" alt="screenshot" src="https://github.com/grafviktor/goto/assets/6890447/1e80a701-cfc4-446c-ae1a-b279cdc5cfc9">

```
Host darkstar
  User dwader
  HostName darkstar.localdomain
  HostkeyAlgorithms +ssh-dss,ssh-rsa
```

<img width="264" alt="screenshot2" src="https://github.com/grafviktor/goto/assets/6890447/3ed298a6-918c-488c-8991-537313a4264b">

Maybe it will also be helpful to check the application log (see `gg -v` to get the log file location) and/or append `-vvv` parameter in the same field where you define the hostname.

What's really annoying is that it's not possible at this moment to import or see your existing ssh config records in GoTo. But i'm planning to support that feature later.

**Comment by tony1661** [comment]
Hi, thanks for the quick response.

I guess I was more wondering if there was plans to have options be entered in the TUI, rather than needing to add them to my ssh config.

The reason being that I would then have only one source of truth for my ssh configs rather than needing to sync multiple files between PCs

**Reply by grafviktor** [reply]
Ah, my apologies! I misunderstood your question 🙃

That's a very interesting topic actually. I had plans adding more ssh parameters. However, after a brief analysis I realized that the quantity of options which are supported by ssh config is really huge.

It will take a tremendous amount of time to support all ssh params using UI components and chances are that these parameters will never be demanded by the majority of users.

At this moment I'm thinking of adding a configuration parameter which would allow to display existing ssh config records in the application, but those records will only be available in read-only mode. 

By doing this, i will increase the intergration with ssh instead of re-implementing the existing functionality. However, I'm still thinking this over and I would be grateful for any ideas. 🙂

**Comment by tony1661** [comment]
For sure, supporting all options that most people do not need may be a wasted effort.

Maybe a way to add custom parameters.

In my case, I would add the parameter as such: `-o HostkeyAlgorithms ssh-dss,ssh-rsa`

This would let users use an infinite amount of variables and your application really only needs to worry about the basics.

Just a though. I do really like the application so far.

**Reply by grafviktor** [reply, marked as answer]
> For sure, supporting all options that most people do not need may be a wasted effort.
> 
> Maybe a way to add custom parameters.
> 
> In my case, I would add the parameter as such: `-o HostkeyAlgorithms ssh-dss,ssh-rsa`
> 

You can already do that. Just append this parameter to your hostname in the `Host` field.

**Comment by tony1661** [comment]
On second thought, would it be worth while just using the ssh config file as the file that stores all the info? Just a thought 🙂

**Reply by grafviktor** [reply]
> On second thought, would it be worth while just using the ssh config file as the file that stores all the info? Just a thought 🙂

That's a great idea mate💡Thank you! I need to consider how to implement that properly 🙂
